tidy3d/components/monitor.py

```diff
@@ -279,7 +279,7 @@ class AbstractModeMonitor(PlanarMonitor, FreqMonitor):
     """:class:`Monitor` that records mode-related data."""
 
     mode_spec: ModeSpec = pydantic.Field(
-        ...,
+        default_factory=ModeSpec,
```

Comment:
Should I change it or do we want to check how the docs look? Generally I would've thought that it's best to instantiate defaults (if they are classes) only on instantiation of the class, otherwise 1. you get object creation on import, which slows imports down and 2. it leads to shared state between different mode monitor instances. All new mode monitors will share the same default instance of `mode_spec`, and if you happen to make a change that modifies this default instance (let's say you write something into `attrs`), then this change will affect all other mode monitors too.

Reply:
i guess we could see how the docs looks, just might require some modification of [this](https://github.com/flexcompute/tidy3d/blob/develop/tidy3d/components/base.py#L1056-L1152)  method that sets the docstring, we'd need to add handling for `defaultfactory`. Also cc'ing @daquinteroflex just so he knows since he's working on a sphinx plugin related to this.

Reply:
I took a stab at this in 1eafd9d which I think would correctly handle cases where we just use a class as a default factory. It wouldn't handle lambdas nicely, although if necessary I can also add support for that.